src/app/cras/edit-cra/edit-cra.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, Params, UrlSegment } from '@angular/router';
+
+import { SerializerService } from 'src/app/shared/serialization/serializer.service';
+import { Cra } from 'src/app/shared/cra.model';
+
+@Component({
+  selector: 'app-edit-cra',
+  templateUrl: './edit-cra.component.html',
+  styleUrls: ['./edit-cra.component.scss']
+})
+export class EditCraComponent implements OnInit {
+
+  cra = new Cra();
+  editToken: string;
+  reviewToken: string;
+
+  saved = false;
+  showModal = false;
+
+  title = {
+    add: 'Créer',
+    edit: 'Editer',
+    review: 'Consulter',
+  };
+
+  mode: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private serializer: SerializerService
+  ) {}
+
+  ngOnInit() {
+
+    this.route.queryParams.subscribe(
+      (parms: Params) => {
+        if (parms.hasOwnProperty('data')) {
+          const datas = <{mode: string, cra: Cra}>this.serializer.deserialize(parms['data']);
```

Comment:
It's better to type the `data` const here instead of use `Type Assertion` on each `datas` property.

Create a `@types` folder with a file like `formData.ts` or something else : 
```
export type formData = {
  mode: string;
  cra: Cra // You will maybe need a `Cra` type too, models in typescript are not types.
};
```

Then you can write : 
```
const datas: formData = this.serializer.deserialize(parms['data']);
```

The goal is to use less as possible the `type assertion`
(cf. https://codecraft.tv/courses/angular/es6-typescript/types/#_type_assertion)